scripts/configure

```diff
@@ -26,98 +14,194 @@ check_dependencies () {
+TOR_PASS=$("./scripts/rpcauth.py" "itdoesntmatter" | tail -1)
+TOR_HASHED_PASS=$(docker run --rm getumbrel/tor:v0.4.1.9 --quiet --hash-password "$TOR_PASS")
+
+
+##########################################################
+### Update config files with configuration variables #####
+##########################################################
+
+if [ "$BITCOIN_CHAIN" == "testnet" ]; then
+  # Set testnet ports
+  BITCOIN_RPC_PORT=18332
+  BITCOIN_P2P_PORT=18333
+  # Uncomment "test" block
+  $gnused -i "s/\#\[test\]/\[test\]/g;" "$BITCOIN_CONF_FILE"
+  # Enable testnet
+  $gnused -i "s/\#testnet=1/testnet=1/g" "$BITCOIN_CONF_FILE"
+  # Switch LND to testnet
+  $gnused -i "s/bitcoin.mainnet=1/bitcoin.testnet=1/g;" "$LND_CONF_FILE"
+  # Comment mainnet neutrino peers
+  $gnused -i "s/neutrino.addpeer=bb2.breez.technology/\;neutrino.addpeer=bb2.breez.technology/g;" "$LND_CONF_FILE"
+  $gnused -i "s/neutrino.addpeer=mainnet1-btcd.zaphq.io/\;neutrino.addpeer=mainnet1-btcd.zaphq.io/g;" "$LND_CONF_FILE"
+  $gnused -i "s/neutrino.addpeer=mainnet2-btcd.zaphq.io/\;neutrino.addpeer=mainnet2-btcd.zaphq.io/g;" "$LND_CONF_FILE"
+  # Uncomment testnet neutrino peers
+  $gnused -i "s/\;neutrino.addpeer=testnet1-btcd.zaphq.io/neutrino.addpeer=testnet1-btcd.zaphq.io/g;" "$LND_CONF_FILE"
+  $gnused -i "s/\;neutrino.addpeer=testnet2-btcd.zaphq.io/neutrino.addpeer=testnet2-btcd.zaphq.io/g;" "$LND_CONF_FILE"
+
 fi
-# if neither set
-if [ -z $TESTNET ] && [ -z $REGTEST ]; then
-    echo "Setting mainnet RPC port in docker-compose"
-    sed -i 's/RPCPORT/8332/g; ' docker-compose.yml
+
+if [ "$BITCOIN_CHAIN" == "regtest" ]; then
+  # Set regtest ports
+  BITCOIN_RPC_PORT=18443
+  BITCOIN_P2P_PORT=18444
+  # Uncomment "regtest" block
+  $gnused -i "s/\#\[regtest\]/\[regtest\]/g;" "$BITCOIN_CONF_FILE"
+  # Enable regtest
+  $gnused -i "s/\#regtest=1/regtest=1/g" "$BITCOIN_CONF_FILE"
+  # Switch LND to regtest
+  $gnused -i "s/bitcoin.mainnet=1/bitcoin.regtest=1/g;" "$LND_CONF_FILE"
+  # Use bitcoind as the node
+  $gnused -i "s/bitcoin.node=neutrino/bitcoin.node=bitcoind/g;" "$LND_CONF_FILE"
 fi
 
+# Update RPC and P2P Ports
+$gnused -i "s/rpcport=<port>/rpcport="$BITCOIN_RPC_PORT"/g;" "$BITCOIN_CONF_FILE"
+$gnused -i "s/port=<port>/port="$BITCOIN_P2P_PORT"/g;" "$BITCOIN_CONF_FILE"
+$gnused -i "s/BITCOIN_RPC_PORT=<port>/BITCOIN_RPC_PORT=""$BITCOIN_RPC_PORT""/g;" "$ENV_FILE"
+$gnused -i "s/BITCOIN_P2P_PORT=<port>/BITCOIN_P2P_PORT=""$BITCOIN_P2P_PORT""/g;" "$ENV_FILE"
+
+# Add rpcauth to bitcoin.conf
+$gnused -i "s/rpcauth=<rpcauth>/"$BITCOIN_RPC_AUTH"/g;" "$BITCOIN_CONF_FILE"
+# Add RPC credentials to lnd.conf
+$gnused -i "s/bitcoind.rpcuser=<username>/bitcoind.rpcuser="$BITCOIN_RPC_USER"/g;" "$LND_CONF_FILE"
+$gnused -i "s/bitcoind.rpcpass=<password>/bitcoind.rpcpass="$BITCOIN_RPC_PASS"/g;" "$LND_CONF_FILE"
+# Add RPC credentials to env file
+$gnused -i "s/BITCOIN_RPC_USER=<username>/BITCOIN_RPC_USER=""$BITCOIN_RPC_USER""/g;" "$ENV_FILE"
```

Comment:
I've noticed this in a few places too.

You're unquoting the string and then re-quoting the variable. This isn't any different to just having the variable directly in the string without breaking the quotes.

I'll submit a PR to your branch fixing all these cases.